glusterd2/brick/glusterfsd.go

```diff
@@ -58,9 +58,14 @@ func (b *Glusterfsd) Args() []string {
 
 	logFile := path.Join(config.GetString("logdir"), "glusterfs", "bricks", fmt.Sprintf("%s.log", brickPathWithoutSlashes))
 
-	brickPort := strconv.Itoa(pmap.AssignPort(0, b.brickinfo.Path))
-
-	volFileID := b.brickinfo.VolfileID + "." + gdctx.MyUUID.String() + "." + brickPathWithoutSlashes
+	portBrickServer := pmap.RegistrySearch(b.brickinfo.Path, pmap.GfPmapPortBrickserver)
+	var brickPort string
+	if portBrickServer == 0 {
+		brickPort = strconv.Itoa(pmap.AssignPort(0, b.brickinfo.Path))
+	} else {
+		brickPort = strconv.Itoa(portBrickServer)
+	}
+	volFileID := b.brickinfo.VolumeName + "." + gdctx.MyUUID.String() + "." + brickPathWithoutSlashes
```

Comment:
The way volfileID should start with brickinfo.VolfileID, please see this https://github.com/gluster/glusterd2/pull/1212/commits/6abc58f5313303bc4aae8f5ae920654e6956ad09#diff-ba737079587260ee0894c46dd5f31916L63 . So you need to replace
:/s/b.brickinfo.VolumeName/b.brickinfo.VolfileID

Reply:
Oh okay. Will make the changes.